### PR TITLE
Compile time parsing (Brush, RelativePoint)

### LINKED ISSUE #5392: Compile Parse methods into IL
It reproduces at `CanvasPage` in ControlCatalog sample app. I suggest the following optimizations:

`RelativePoint.Parse("100%,100%");` -> `new RelativePoint(1.0, 1.0, RelativeUnit.Absolute)`
`Point.Parse("0,0")` -> `new Point(0, 0)`
`Color.Parse("Black")` -> `new Color(1, 0, 0, 0)`

And other objects declared in `Avalonia.Media`.

It is also possible to get rid of some excess invocations if use caching of immutable objects.

### PR DESCRIPTION
**Pull request by KvanTTT**
## What does the pull request do?

Adds compile-time parsing for brushes and other classes.

## What is the current behavior?

Runtime parsing.

## What is the updated/expected behavior with this PR?

Parsing during compilation.

### Before

```cs
new BrushConverter().ConvertFrom((ITypeDescriptorContext)(object)context, CultureInfo.InvariantCulture, "Blue")
```

### After

```cs
KnownColors.ToBrush(Avalonia.Media.KnownColor.Blue)
```

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Fixed issues

Fixes #5392

### CODE REVIEW

**Comment by MarchingCube** [comment]
I guess this is mostly a side-effect of how `Brush.Parse` works but now I'm wondering - what was the reasoning behind:
- `Brush.Parse("Red")` returning an immutable brush
- `Brush.Parse("#FF0000")` returning a mutable brush

Since one potential optimization in this case would be to refer to `Brushes.Red` which would save us a bunch of allocations.
@grokys

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AstNodes/AvaloniaXamlIlVisualElementAstNode.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+using Avalonia.Media;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.TypeSystem;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.AstNodes
+{
+    class AvaloniaXamlIlVisualElementAstNode<T> : XamlAstNode, IXamlAstValueNode, IXamlAstILEmitableNode
```
Comment (kekekeks) [inline src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AstNodes/AvaloniaXamlIlVisualElementAstNode.cs:9]: I think we already have an AST node for creating new objects

File: src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj [hunk 2]
```diff
@@ -90,9 +90,30 @@
       <Compile Include="../Avalonia.Controls/GridLength.cs">
         <Link>Markup/%(RecursiveDir)%(FileName)%(Extension)</Link>
       </Compile>
+      <Compile Include="../Avalonia.Visuals/RelativePoint.cs">
+        <Link>Markup/%(RecursiveDir)%(FileName)%(Extension)</Link>
+      </Compile>
+      <Compile Include="../Avalonia.Visuals/Media/Brush.cs">
```
Comment (kekekeks) [inline src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:96]: That's quite a lot of files and quite a bit of changes that could be replaced by just parsing the color.

**Review by kekekeks — CHANGES_REQUESTED** [review]

File: src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AstNodes/AvaloniaXamlIlVisualElementAstNode.cs (hunk 1 above)
Comment (kekekeks) [reply src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AstNodes/AvaloniaXamlIlVisualElementAstNode.cs:9]: `XamlAstNewClrObjectNode` should be sufficient.

File: src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj (hunk 2 above)
Comment (KvanTTT) [reply src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:96]: Should it be added directly to Avalonia.Build.Tasks project?

Comment (kekekeks) [reply src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:96]: I think brush-related stuff isn't needed by compiler at all. We only support converting solid color brushes, so just using existing color parser is enough.

Comment (KvanTTT) [reply src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:96]: But `KnownColors.ToBrush(Avalonia.Media.KnownColor.Blue)` looks better than `new BrushConverter().ConvertFrom((ITypeDescriptorContext)(object)context, CultureInfo.InvariantCulture, "Blue")`

Comment (MarchingCube) [reply src/Avalonia.Build.Tasks/Avalonia.Build.Tasks.csproj:96]: I think idea here is to parse a color to an integer and emit `new SolidColorBrush(parsedColor)` instead. Which does not require referencing new files. We could even consider emitting immutable brushes (although that might not work too well for animations).

**Pull request closed without merging**